resources/ajax_processing/importFromEbscoKb.php

```diff
@@ -0,0 +1,829 @@
+            }
+        }
+    }
+
+    // Workflow and default order
+    if ($newWorkflow && empty($resource->getCurrentWorkflowID())){
+        // Create the default order
+        $resourceAcquisition = new ResourceAcquisition();
+        $resourceAcquisition->resourceID = $resource->primaryKey;
+        $resourceAcquisition->subscriptionStartDate = date("Y-m-d");
+        $resourceAcquisition->subscriptionEndDate = date("Y-m-d");
+        $resourceAcquisition->save();
+        $resource->enterNewWorkflow();
+    }
+    return $resource;
+
+}
+
+function addProvider(Resource $resource){
+    global $organizationId, $organizationRoleId;
+
+    if ($organizationId && $organizationRoleId){
+
+        // create an original list of organzational links
+        $linkedOrganizations = array_map(function($org){
+            return ['organizationId' => $org['organizationID'], 'organizationRoleId' => $org['organizationRoleID']];
+        }, $resource->getOrganizationArray());
+        $linkedOrganizations[] = ['organizationId' => $organizationId, 'organizationRoleId' => $organizationRoleId];
+        $linkedOrganizations = array_map("unserialize", array_unique(array_map("serialize", $linkedOrganizations)));
+        // Remove old links
+        $resource->removeResourceOrganizations();
+        foreach($linkedOrganizations as $link){
+            $resourceOrganizationLink = new ResourceOrganizationLink();
+            $resourceOrganizationLink->resourceID = $resource->primaryKey;
+            $resourceOrganizationLink->organizationID = $link['organizationId'];
+            $resourceOrganizationLink->organizationRoleID = $link['organizationRoleId'];
+            try {
+                $resourceOrganizationLink->save();
+            } catch (Exception $e) {
+                send_errors([create_error('general', 'Could not add resource provider', $e->getMessage())]);
+            }
+        }
+    }
+}
+
+function addNotes(Resource $resource, $notes = []){
+
+    global $loginID;
+
+    if(!empty($notes)){
+
+        $existingNotes = $resource->getNotes();
+        $existingNoteText =  array_map(function($note){ return $note->noteText; }, $existingNotes);
+        //this is just to figure out what the creator entered note type ID is
+        $noteType = new NoteType();
+
+        foreach($notes as $note){
+            // If the note text doesn't already exist, add it
+            if(!in_array($note['text'], $existingNoteText)) {
+
+                $resourceNote = new ResourceNote();
+                $resourceNote->resourceNoteID = '';
+                $resourceNote->updateLoginID = $loginID;
+                $resourceNote->updateDate = date( 'Y-m-d' );
+                $resourceNote->noteTypeID = $noteType->getInitialNoteTypeID();
+                $resourceNote->tabName = $note['tab'];
+                $resourceNote->resourceID = $resource->primaryKey;
```

Comment:
Since multiple ordering, notes can be tied to a resource or to an order. ResourceNote.resourceID has been replaced with ResourceNote.entityID.
More specifically:
 - entityID should contain Resource.resourceID for the Product tab
 - entityID should contain ResourceAcquisition.resourceAcquisitionID for Acquisitions, Access and Cataloging tabs.

Reply:
@veggiematts I'm still a bit lost on this. I'm using the exact same code that is in the [submitNewResource.php](https://github.com/coral-erm/coral/blob/development/resources/ajax_processing/submitNewResource.php#L70) file.

Is there new code I should be looking at?

Reply:
nvm, I think I sorted it out

Reply:
ok, this is resolved as well